runtime/shiden/src/lib.rs

```diff
@@ -740,8 +744,28 @@ pub type Executive = frame_executive::Executive<
     frame_system::ChainContext<Runtime>,
     Runtime,
     AllPallets,
+    (DappsStakingMigrationV2,),
```

Comment:
Is the general rule that if we're doing pallet specific storage migrations, then we use a storage versioning variable in that pallet to make sure we don't run migrations twice? I had only seen these migrations happen once before here, and the spec version was checked since the storage change was not for a specific pallet: https://github.com/paritytech/polkadot/pull/4655/files#diff-e5e76e02c0d16e79c70b024cbe3c6ea56f3249382a0f987ba203c34fcb40ed66R1775

Reply:
Storage versioning variable is used by many different pallets so I think we could say it is a general rule.

If you're wondering why we don't just use pallet hooks, check this comment:
https://github.com/AstarNetwork/Astar/pull/509#discussion_r757788350

Reply:
That link doesn't go anywhere (unless I'm missing some permissions). I see the [commit](https://github.com/AstarNetwork/Astar/pull/509/commits/d91c2175e1c4f155af9527ac348b4d618e1793dd) where you moved the migration hooks from the pallet into runtime hooks. I'm sure there is reasoning for this; can you elaborate? To me it just looks like WET code. I see the storage version checks get moved to inside the migration function.

Reply:
You can hover over the link to see the comment.
Or scroll up and find comment from akru made on Nov 27, 2021.

Not sure what exactly are you referring to with WET. The version check is only in one place. Or?

Reply:
Awesome, I see the comment now and everything makes sense.

By WET I meant "Write Everything Twice"; I was confused as to why migrations were rewritten in each runtime instead of written in the pallet and called in the runtime, but thats because we're not using pallet hooks :)